src/components/button/button.test.js

```diff
@@ -0,0 +1,191 @@
+import * as React from "react";
+import Button from "./button.component";
+
+import {
+  buttonSubtextPreview,
+  buttonDataComponent,
+} from "../../../cypress/locators/button";
+
+import { icon, tooltipPreview } from "../../../cypress/locators";
+import { positionOfElement, keyCode } from "../../../cypress/support/helper";
+import CypressMountWithProviders from "../../../cypress/support/component-helper/cypress-mount";
+
+const testData = ["mp150ú¿¡üßä", "!@#$%^*()_+-=~[];:.,?{}&\"'<>"];
+
+const ButtonDifferentTypes = ({ ...props }) => {
+  return (
+    <div>
+      <Button buttonType="primary" {...props}>
+        Primary
+      </Button>
+      <Button buttonType="secondary" {...props}>
+        Secondary
+      </Button>
+      <Button buttonType="tertiary" {...props}>
+        Tertiary
+      </Button>
+      <Button buttonType="dashed" {...props}>
+        Dashed
+      </Button>
+    </div>
+  );
+};
+
+context("Test for Button component", () => {
+  describe("Check props for Button component", () => {
+    it.each(testData)(
+      "should render Button label using %s special characters",
+      (label) => {
+        CypressMountWithProviders(<Button>{label}</Button>);
+
+        buttonDataComponent().should("have.text", label);
+      }
+    );
+
+    it.each(testData)(
+      "should render Button subtext with %s special characters",
+      (subtext) => {
+        CypressMountWithProviders(<Button size="large" subtext={subtext} />);
+
+        buttonSubtextPreview().should("have.text", subtext);
+      }
+    );
+
+    it.each(testData)(
+      "should render tooltip message with %s special characters",
+      (tooltipmessage) => {
+        CypressMountWithProviders(
+          <Button
+            iconType="bin"
+            iconTooltipMessage={tooltipmessage}
+            m="100px"
+          />
+        );
+
+        buttonDataComponent().children().last().realHover();
+        tooltipPreview().should("have.text", tooltipmessage);
+      }
+    );
+
+    it.each([
+      ["after", "left"],
+      ["before", "right"],
+    ])(
+      "should set position to %s for icon in a button",
+      (iconPosition, margin) => {
+        CypressMountWithProviders(
+          <Button iconType="add" iconPosition={iconPosition}>
+            IconPosition
+          </Button>
+        );
+
+        icon().should("have.css", `margin-${margin}`, "8px");
+      }
+    );
+
+    it("should render Button with full width", () => {
+      CypressMountWithProviders(<Button fullWidth />);
+
+      buttonDataComponent().should("have.css", "width").and("eq", "1350px");
```

Comment:
Why does the width need to be specified in number of pixels for a full width button? Shouldn't it just be a percentage? I'd have thought that the number of pixels width will differ depending on viewport size.